drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -914,13 +892,14 @@ static int siwx91x_set_power_save(const struct device *dev, struct wifi_ps_param
 			sl_ps_profile->dtim_aligned_type = 1;
 			sl_ps_profile->listen_interval = 0;
 			bcn_li = 0;
+			sidev->listen_interval_wakeup = false;
 		} else if (params->wakeup_mode == WIFI_PS_WAKEUP_MODE_LISTEN_INTERVAL) {
 			sl_ps_profile->dtim_aligned_type = 0;
 
 			if (sl_ps_profile->profile != HIGH_PERFORMANCE) {
 				status = siwx91x_schedule_listen_interval_wakeup(dev, bcn_li);
-				if (status < 0) {
-					return status;
+				if (status == -EINVAL) {
+					sl_ps_profile->profile = HIGH_PERFORMANCE;
```

Comment:
1. Is it related to this commit?
2. `EINVAL` means "Invalid argument". This is a bit weird to fix this error by changing the profile.
3. What happens now if `status < 0 &&  status != -EINVAL` ?